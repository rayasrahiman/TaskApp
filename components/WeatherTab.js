import React from 'react';
import {View, Text, StyleSheet} from 'react-native';
import {
  widthPercentageToDP as wp,
  heightPercentageToDP as hp,
} from 'react-native-responsive-screen';
import FontAwesomeIcon from 'react-native-vector-icons/FontAwesome5';
import Ionicon from 'react-native-vector-icons/Ionicons';
import FeatherIcon from 'react-native-vector-icons/Feather';

import {Colors} from '../constants/colors';
import I18n from '../languages/i18n';

export default function WeatherTab() {
  return (
    <View style={styles.container}>
      <View style={styles.firstCont}>
        <FontAwesomeIcon name="cloud-sun" style={styles.iconSize} />
        <Text style={styles.titleTxt}>{I18n.t("PartlyCloudy")}</Text>
      </View>
      <View style={styles.secondMainCont}>
        <View style={styles.cont}>
          <Ionicon name="water" style={styles.iconSize} />
          <Text style={styles.titleTxt}>{I18n.t("78")}</Text>
        </View>
        <View style={styles.cont}>
          <FeatherIcon name="wind" style={styles.iconSize} />
          <Text style={styles.titleTxt}>{I18n.t("70")}</Text>
        </View>
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    padding: hp(3),
    backgroundColor: Colors.lightestBlue,
    shadowColor: Colors.grayishBlack,
    shadowOffset: {
      width: 0,
      height: 0.3,
    },
    shadowOpacity: 0.25,
    elevation: 5,
  },
  firstCont: {
    marginHorizontal: wp(0.5),
    flexDirection: 'row',
    alignItems: 'center',
  },
  titleTxt: {
    fontWeight: 'bold',
    fontSize: hp(3),
    color: Colors.darkBlue,
  },
  secondMainCont: {
    flexDirection: 'row',
  },
  cont: {
    marginHorizontal: wp(3),
    flexDirection: 'row',
    alignItems: 'center',
  },
  iconSize: {
    fontSize: hp(4),
    marginRight: wp(2),
    color: Colors.darkBlue,
  },
});
