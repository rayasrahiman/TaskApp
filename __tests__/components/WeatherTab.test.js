import React from 'react';
import {fireEvent, render, screen} from '@testing-library/react-native';
import {shallow} from 'enzyme';

import WeatherTab from '../../components/WeatherTab';

beforeEach(jest.spyOn(console, 'error').mockImplementation(() => undefined));
afterEach(() => jest.clearAllMocks());

describe('WeatherTab', () => {
  test('should render WeatherTab correctly', () => {
    const wrapper = render(<WeatherTab />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
