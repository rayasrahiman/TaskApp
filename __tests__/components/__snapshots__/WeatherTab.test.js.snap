// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherTab should render WeatherTab correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#EBF0F4",
      "elevation": 5,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "padding": 40,
      "shadowColor": "#838383",
      "shadowOffset": Object {
        "height": 0.3,
        "width": 0,
      },
      "shadowOpacity": 0.25,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "marginHorizontal": 4,
      }
    }
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 12,
          },
          Object {
            "color": "#244379",
            "fontSize": 53.5,
            "marginRight": 15,
          },
          Object {
            "fontFamily": "FontAwesome5Free-Solid",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {
            "fontWeight": "900",
          },
        ]
      }
    >
      
    </Text>
    <Text
      style={
        Object {
          "color": "#244379",
          "fontSize": 40,
          "fontWeight": "bold",
        }
      }
    >
      Partly Cloudy
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginHorizontal": 22.5,
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Object {
              "color": "#244379",
              "fontSize": 53.5,
              "marginRight": 15,
            },
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
      <Text
        style={
          Object {
            "color": "#244379",
            "fontSize": 40,
            "fontWeight": "bold",
          }
        }
      >
        78°F
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginHorizontal": 22.5,
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Object {
              "color": "#244379",
              "fontSize": 53.5,
              "marginRight": 15,
            },
            Object {
              "fontFamily": "Feather",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
      <Text
        style={
          Object {
            "color": "#244379",
            "fontSize": 40,
            "fontWeight": "bold",
          }
        }
      >
        70°F
      </Text>
    </View>
  </View>
</View>
`;
